include/swift/AST/DiagnosticsParse.def

```diff
@@ -727,10 +727,9 @@ ERROR(generic_non_function,PointsToFirstBadToken,
 ERROR(rethrowing_function_type,none,
       "only function declarations may be marked 'rethrows'; "
       "did you mean 'throws'?", ())
-ERROR(throws_in_wrong_position,none,
-      "'throws' may only occur before '->'", ())
-ERROR(rethrows_in_wrong_position,none,
-      "'rethrows' may only occur before '->'", ())
+ERROR(async_or_throws_in_wrong_position,none,
```

Comment:
Hi @DougGregor – Congrats on the first part of these change. A small nit here. It would be nice if the diagnostic selector aligned with the diagnostic name. Right now it's backwards. Specifically, the diag is named "async…" and then "…or_throws" but the first selector is "throws". The fix would be:
```
ERROR(async_or_throws_in_wrong_position,none,
      "%select{'async'|'throws'|'rethrows'}0 may only occur before '->'",
       (unsigned))
```